lab-taylor/app/html/index.html

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
+<html ng-app="cowsayApp">
+  <head>
+    <meta charset="utf-8">
+    <title>__cowsay-client__</title>
+    <script src="bundle.js"></script>
+    <link rel="stylesheet" type="text/css" href="bundle.css" />
+  </head>
+  <body>
+    <main ng-controller="CowsayController">
+      <form ng-submit="say()">
+        <label for="cowsay-text">Make the Cow Say Something<br /></label>
+        <input type="cowsay-text" text="text" ng-model="cowsayText">
+
+        <select name="cowsayFile" id="cowsayFile" ng-model="cowsayFile">
+          <option ng-repeat="cow in cowFiles" value="{{cow}}">{{ cow }}</option>
+        </select>
+
+        <br />
+        <button type="cowsay-button" name="cowsay-button" ng-click="finalize(cowsayText, cowsayFile)">Finalize Cow</button>
```

Comment:
Button types can be "button", "submit", or "reset".
The finalize button should have type="submit". You can then remove ng-click since finalize() is being called by the form on submit using the ng-submit directive.